traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java

```diff
@@ -62,4 +60,81 @@ public void onFlightSearchError(Error error) {
         Assert.assertEquals("Error(s) occurred while querying API for flight 'AC100'!\n" +
                 "See logcat for more info\n", 0, errorList.size());
     }
+
+    @Test
+    public void catalogueTest() throws Exception {
+        CountDownLatch latch = new CountDownLatch(1);
+        List<Flight> flights = new ArrayList<>();
+        final List<Error> errorList = new ArrayList<>();
+
+        CatalogQuery catalogueQuery = new CatalogQuery(flights);
+        Traveler.fetchCatalog(catalogueQuery, new CatalogSearchCallback() {
+            @Override
+            public void onCatalogSuccess(Catalog catalog) {
+                TravelerLog.v("TravelerTest", "catalogueTest::onCatalogSuccess");
+                TravelerLog.v("TravelerTest", catalog.toString());
+                latch.countDown();
+            }
+
+            @Override
+            public void onCatalogError(Error error) {
+                TravelerLog.e("TravelerTest", "catalogueTest::onCatalogError");
+                error.printStackTrace();
+                errorList.add(error);
+                latch.countDown();
+            }
+        });
+
+        latch.await();
+        Assert.assertEquals("Error(s) occurred while querying catalogue with no flights\n", 0, errorList.size());
+    }
+
+    /**
+     * This test can only succeed if flightSearchTest had retrieved a valid flight ID
+     */
+    @Test
+    public void flightSearchWithCatalogueTest() throws Exception {
+        Date date = Calendar.getInstance().getTime();
+        FlightQuery query = new FlightQuery("AC100", date);
+
+        CountDownLatch latch = new CountDownLatch(1);
+        final List<Error> errorList = new ArrayList<>();
+
+        Traveler.flightSearch(query, new FlightSearchCallback() {
```

Comment:
No need for this chained calls, should create a mock Flight on the spot and use that

Reply:
A mocked flight would need a valid flight ID, The flight IDs used to call the API are constantly changing, and Brian's data cannot query the past.  lets chat about this one

Reply:
updated comment